app/src/main/java/com/raghav/spacedawn/ui/LoadStateViewHolder.kt

```diff
@@ -2,29 +2,32 @@ package com.raghav.spacedawn.ui
 
 import android.view.LayoutInflater
 import android.view.ViewGroup
-import androidx.core.view.isVisible
 import androidx.paging.LoadState
 import androidx.recyclerview.widget.RecyclerView
 import com.raghav.spacedawn.R
 import com.raghav.spacedawn.databinding.ItemLoadStateBinding
+import com.raghav.spacedawn.ui.common.ItemLoadState
 
 class LoadStateViewHolder(
     private val binding: ItemLoadStateBinding,
-    retryListener: () -> Unit
+    private val retryListener: () -> Unit
 ) :
     RecyclerView.ViewHolder(binding.root) {
 
-    init {
-        binding.btnRetry.setOnClickListener { retryListener.invoke() }
-    }
-
     fun bind(loadState: LoadState) {
-        if (loadState is LoadState.Error) {
-            binding.tvErrorMessage.text = loadState.error.localizedMessage
+        binding.composeView.setContent {
+            ItemLoadState(
+                message = if (loadState is LoadState.Error) {
+                    loadState.error.localizedMessage
+                } else {
+                    ""
+                },
+                isCtaVisible = loadState is LoadState.Error,
+                isProgressBarVisible = loadState is LoadState.Loading
+            ) {
+                retryListener()
+            }
```

Comment:
If the code refactor is not too great, then it is generally advised to pick a side (i.e. either go full Compose or stay on XML) because of the drastic performance hit we get when we use Composables inside a RecyclerView.

Ref:
1. https://medium.com/mobile-app-development-publication/recyclerview-with-compose-view-is-not-performant-97c74df4b77f
2. https://engineering.premise.com/measuring-render-performance-with-jetpack-compose-c0bf5814933

Reply:
Thanks for the references I'll go through them, the next step is indeed to use a Composable function such as Lazy Column to display scrollable lists. The aim of this exercise is to get practice with ComposeView.